docs/api_guide/plot_06_sklearn_pipeline_cv_demo.py

```diff
@@ -0,0 +1,473 @@
+"""
+# Selecting basis by cross-validation with scikit-learn
+
+In this demo, we will demonstrate how to select an appropriate basis and its hyperparameters using cross-validation.
+In particular, we will learn:
+
+1. What a scikit-learn pipeline is.
+2. Why pipelines are useful.
+3. How to combine NeMoS `Basis` and `GLM` objects in a pipeline.
+4. How to select the number of bases and the basis type through cross-validation (or any other hyperparameter in the pipeline).
+5. How to use a custom scoring metric to quantify the performance of each configuration.
+
+"""
+
+# %%
+# ## What is a scikit-learn pipeline
+#
+# <figure markdown>
+# <img src="../../../assets/pipeline.svg" style="width: 100%"/>
+# <figcaption>Schematic of a scikit-learn pipeline.</figcaption>
+# </figure>
+#
+# A pipeline is a sequence of data transformations. Each step in the pipeline transforms the input data into a different representation. The final step is either another transformation or a model step that fits, predicts, or scores based on the previous step's output and some observations. Setting up such machinery can be simplified using the `Pipeline` class from scikit-learn.
```

Comment:
```suggestion
# A pipeline is a sequence of data transformations leading up to a model. Each step before the final one transforms the input data into a different representation, and then the final model step fits, predicts, or scores based on the previous step's output and some observations. Setting up such machinery can be simplified using the `Pipeline` class from scikit-learn.
```

While this is more limited than the pipelines actually are, I think it's clearer (and true for our use cases)